types/react-intl/index.d.ts

```diff
@@ -11,6 +11,10 @@
 declare namespace ReactIntl {
 
     interface Values {
+        [key: string]: string | JSX.Element;
```

Comment:
I now inlined the type definitions. I guess it looks a bit clearer, esp. since now it doesn't hint that formatMessage and formatHTMLMessage take in the same kind of string->string mapping.